packages/graphql/src/translate/queryAST/factory/OperationFactory.ts

```diff
@@ -72,17 +73,48 @@ export class OperationsFactory {
         entity: ConcreteEntityAdapter | RelationshipAdapter | InterfaceEntityAdapter | UnionEntityAdapter,
         resolveTree: ResolveTree,
         context: Neo4jGraphQLTranslationContext
-    ): ReadOperation | CreateOperation {
-        if (isConcreteEntity(entity) && resolveTree.name === entity.operations.rootTypeFieldNames.create) {
-            return this.createUnwindCreateOperation(entity, resolveTree, context);
+    ): Operation {
+        if (isConcreteEntity(entity)) {
+            const operationMatch = parseOperationField(resolveTree.name, entity);
+            if (operationMatch.isCreate) {
+                return this.createCreateOperation(entity, resolveTree, context);
+            } else if (operationMatch.isRead) {
+                const op = this.createReadOperation(entity, resolveTree, context) as ReadOperation;
+                op.nodeAlias = TOP_LEVEL_NODE_NAME;
+                return op;
+            } else if (operationMatch.isConnection) {
+                const topLevelConnectionResolveTree = this.fixResolveTreeForTopLevelConnection(resolveTree);
+                const op = this.createConnectionOperationAST(
+                    entity,
+                    topLevelConnectionResolveTree,
+                    context
+                ) as ConnectionReadOperation;
+                op.nodeAlias = TOP_LEVEL_NODE_NAME;
+                return op;
+            }
+            throw new Error("Not implemented");
         }
-
-        const op = this.createReadOperation(entity, resolveTree, context) as ReadOperation;
-        op.nodeAlias = TOP_LEVEL_NODE_NAME;
-        return op;
+        return this.createReadOperation(entity, resolveTree, context);
```

Comment:
```suggestion
if (!isConcreteEntity(entity)) {
        return this.createReadOperation(entity, resolveTree, context);
    }
    const operationMatch = parseOperationField(resolveTree.name, entity);
    if (operationMatch.isCreate) {
        return this.createCreateOperation(entity, resolveTree, context);
    }  
    if (operationMatch.isRead) {
        const op = this.createReadOperation(entity, resolveTree, context) as ReadOperation;
        op.nodeAlias = TOP_LEVEL_NODE_NAME;
        return op;
    }  
    if (operationMatch.isConnection) {
        const topLevelConnectionResolveTree = this.fixResolveTreeForTopLevelConnection(resolveTree);
        const op = this.createConnectionOperationAST(
            entity,
            topLevelConnectionResolveTree,
            context
        ) as ConnectionReadOperation;
        op.nodeAlias = TOP_LEVEL_NODE_NAME;
        return op;
    }
    throw new Error("Not implemented");  
```

Reply:
this nested if check doesn't look very easy to understand. We can return early and remove the `else` as each if-case returns. We could also consider using a switch-case to understand the set of all possibilities?